modulus/models/module.py

```diff
@@ -48,8 +73,23 @@ def __init__(self, meta: ModelMetaData = None):
         self.logger.addHandler(handler)
         self.logger.setLevel(logging.WARNING)
 
-        # dummy buffer for getting where the networks device
-        self.register_buffer("device_buffer", torch.empty(0))
+    @classmethod
+    def instantiate(cls, arg_dict):
+        _mod = importlib.import_module(arg_dict["__module__"])
+        _cls_name = arg_dict["__name__"]
+
+        # Add a check if the class is one of the dynamically created ones
+        if _cls_name in cls._dynamically_created_classes:
+            _cls = cls._dynamically_created_classes[_cls_name]
+        else:
+            _cls = getattr(_mod, _cls_name)
```

Comment:
Maybe you could you use your factory method here? I suppose _cls_name is the direct class name, not the entry point key, but consistency in how we initialize things would be nice if possible.

Reply:
Fixed this